### PR TITLE
[Feature] Implement permission handling

### PR DESCRIPTION
**Pull request by luongvo**
## What happened 🤔
- Continue #50 to implement permission handling

## Insight 👀
- Add [RxPermissions](https://github.com/tbruyelle/RxPermissions) to handle `CAMERA` permission on Second screen
- Clean up + combine `SchedulerProviderModule` into `AppModule`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -38,14 +38,14 @@ buildscript {
...
 
-        permission_dispatcher_version = '4.3.0'
         play_service_version = '12.0.1'
 
         retrofit_version = '2.4.0'
         rxandroid_version = '2.0.1'
         rxbinding_version = "3.1.0"
         rxjava_version = '2.1.4'
         rxkotlin_version = '2.2.0'
+        rxpermission_version = "0.10.2"
```
Comment (manh-t) [inline build.gradle:48]: @luongvo should we update to latest version `0.12` [Rxpermission](https://github.com/tbruyelle/RxPermissions)

Comment (luongvo) [reply build.gradle:48]: @manh-t Rxpermission 0.12 requires rxjava3 as we're using rxjava2, it should be upgraded later

**Review by manh-t — APPROVED** [review]

File: app/src/main/java/co/nimblehq/di/modules/FragmentComponentsModule.kt [hunk 2]
```diff
@@ -14,3 +17,11 @@ abstract class NavigatorModule {
     @Binds
     abstract fun mainNavigator(mainNavigator: MainNavigatorImpl): MainNavigator
 }
+
+@InstallIn(FragmentComponent::class)
+@Module
+class RxPermissionsModule {
+
+    @Provides
+    fun rxPermission(fragment: Fragment): RxPermissions = RxPermissions(fragment)
```
Comment (Tuubz) [inline app/src/main/java/co/nimblehq/di/modules/FragmentComponentsModule.kt:26]: No biggie: `rxPermissions`

Same goes for `lateinit var rxPermissions` & `rxpermissions_version`

File: app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt [hunk 3]
```diff
@@ -34,29 +39,28 @@ class SecondFragment : BaseFragment() {
...
+            permission.granted -> {
+                startActivity(Intent(MediaStore.ACTION_IMAGE_CAPTURE))
+            }
+            permission.shouldShowRequestPermissionRationale -> {
+                // Deny
+                toaster.display("Permission camera denied")
+            }
+            else -> {
+                // Deny and never ask again
+                toaster.display("Permission camera never ask")
```
Comment (Tuubz) [inline app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt:60]: No biggie: "Permission camera never ask again"

**Review by Tuubz — APPROVED** [review]

File: app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt (hunk 3 above)
Comment (minhnimble) [reply app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt:60]: or `Camera permission is never asked again` 😁

**Review by minhnimble — APPROVED** [review]

**Comment by luongvo** [comment]
@hoangnguyen92dn please manage to merge this PR 🙏

File: app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt [hunk 4]
```diff
@@ -34,29 +39,28 @@ class SecondFragment : BaseFragment() {
...
+                startActivity(Intent(MediaStore.ACTION_IMAGE_CAPTURE))
+            }
+            permission.shouldShowRequestPermissionRationale -> {
+                // Deny
+                toaster.display("Permission camera denied")
+            }
+            else -> {
+                // Deny and never ask again
+                toaster.display("Permission camera never ask again")
+            }
```
Comment (hoangnguyen92dn) [inline app/src/main/java/co/nimblehq/ui/screens/second/SecondFragment.kt:61]: Not a big deal but we could get rid of brackets :D

**Review by hoangnguyen92dn — APPROVED** [review]

**Pull request merged**